overlord/ifacestate/handlers.go

```diff
@@ -1217,3 +1217,63 @@ func (m *InterfaceManager) doGadgetConnect(task *state.Task, _ *tomb.Tomb) error
 	task.SetStatus(state.DoneStatus)
 	return nil
 }
+
+// doHotplugDisconnect creates task(s) to disconnect connections and remove slots in response to hotplug "remove" event.
+func (m *InterfaceManager) doHotplugDisconnect(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	syssnap, err := systemSnapInfo(st)
+	if err != nil {
+		return err
+	}
+
+	ifaceName, hotplugKey, err := getHotplugAttrs(task)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get hotplug task attributes: %s", err)
+	}
+
+	connections, err := m.repo.ConnectionsForHotplugKey(ifaceName, hotplugKey)
+	if err != nil {
+		return err
+	}
+	if len(connections) == 0 {
+		return nil
+	}
+
+	// check for conflicts on all connections first before creating disconnect hooks
+	for _, connRef := range connections {
+		if err := checkDisconnectConflicts(st, syssnap.InstanceName(), connRef.PlugRef.Snap, connRef.SlotRef.Snap); err != nil {
```

Comment:
not sure setting/passing disconnectingSnap looks right:

// another task related to same snap op (unrelated op would be blocked by snapstate conflict logic)
                if otherSnapName == disconnectingSnap {
                        continue
                }

that bit of reasoning doesn't apply here I think, we create this kind of changes eagerly, we don't check for conflicts at all in removeDevice itself

core might be refreshing, would have still created the task if the device was going away, but we do need to wait here.

We need some tests about conflict checks and scenarios.